File: benchmarks/micro/TextLiteralLengthPruning.c
#include <stdint.h>
#include <string.h>
#include <unistd.h>

static int64_t score(int64_t salt) {
    if (strlen("stark-performance") == 17) {
        salt = (salt + 51) % 1000003;
    } else {
        salt -= 100000;
    }

    if ((sizeof("llvm-output") - 1) != 11) {
        return salt - 77777;
    }

    return ((salt + 11) * 17) % 1000003;
}

int main(void) {
    int64_t total = (int64_t)getpid();

    for (int32_t i = 0; i < 200000; i += 1) {
        total += score(total + (int64_t)i);
        total %= 1000000007;
    }

    return total <= 0 ? 1 : 0;
}

File: benchmarks/micro/TextLiteralLengthPruning.rs
fn score(mut salt: i64) -> i64 {
    if "stark-performance".len() == 17 {
        salt = (salt + 51) % 1_000_003;
    } else {
        salt -= 100_000;
    }

    if "llvm-output".chars().count() != 11 {
        return salt - 77_777;
    }

    ((salt + 11) * 17) % 1_000_003
}

fn main() {
    let mut total = std::process::id() as i64;

    for i in 0_i32..200_000 {
        total += score(total + i as i64);
        total %= 1_000_000_007;
    }

    if total <= 0 {
        std::process::exit(1);
    }
}
